### PR TITLE
Feat(eos_cli_config_gen): Add FIPS and hostkey.client features to management_ssh

### LINKED ISSUE #3248: Management SSH - Missing Features
### Enhancement summary

We would like to be able to set:

fips restrictions
hostkey client strict-checking

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

We use these settings per compliance.

### Describe the solution you would like

Something akin to the following?
```
management_ssh:
  fips: true
  hostkey:
    client:
      strict-checking: true
```

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

- Add FIPS and hostkey.client features to management_ssh.
- Add `server_cert` to hostkey
- Add the molecule test for both.

## Related Issue(s)

Fixes #3248 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## How to test

```
management ssh
   fips restrictions
   hostkey client strict-checking
   hostkey server cert <switch's cert key>
```

### User Checklist

- [x] Verify the config ordering in EOS device.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/management-ssh.yml [hunk 1]
```diff
@@ -15,3 +15,6 @@ management_ssh:
   client_alive:
     count_max: 42
     interval: 666
+  hostkey:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/management-ssh.yml:18]: please also add the hostkey.server... that was already in the schema. Seems like we have missed testing that. Thanks!

**Review by ClausHolbechArista — DISMISSED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/management-ssh.yml (hunk 1 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/management-ssh.yml:18]: Sure, I will add it.

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/management-ssh.yml:18]: Modified the schema and template with test case

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/management-ssh.yml:18]: The test for `hostkey.server` is already there in `management-ssh-custom-cipher.yml`.
I have added `server_cert_key` to the data-model and test-case for it.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-ssh.j2 [hunk 2]
```diff
@@ -51,9 +51,18 @@ management ssh
 {%     if management_ssh.mac is arista.avd.defined %}
    mac {{ management_ssh.mac | join(" ") }}
 {%     endif %}
+{%     if management_ssh.fips_restrictions is arista.avd.defined(true) %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-ssh.j2:57]: Please adjust the order. From my test on EOS:
```eos
management ssh
   ip access-group ACL-SSH-VRF vrf mgt in
   ip access-group ACL-SSH in
   idle-timeout 15
   connection per-host 10
   fips restrictions
   hostkey client strict-checking
   client-alive interval 666
   client-alive count-max 42
   log-level debug
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml [hunk 3]
```diff
@@ -58,14 +58,23 @@ keys:
...
       hostkey:
         type: dict
         keys:
           server:
-            description: SSH host key settings
+            description: SSH host key settings.
             type: list
             items:
               type: str
+          server_cert_key:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/management_ssh.schema.yml]:
```suggestion
          server_cert:
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-ssh.j2 (hunk 2 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-ssh.j2:57]: For me it is showing below `key-exchange` and `hostkey server`.
 `client-alive` and `hostkey server` position is kept as earlier in the template.

I should move it after `hostkey server` or before `client-alive`?

```
management ssh
   key-exchange ecdh-sha2-nistp521
   hostkey server dsa
   fips restrictions
   hostkey client strict-checking
   client-alive count-max 10
```

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-ssh.j2:57]: I think the sequence like:
1. fips restrictions
2. hostkey client strict-checking
3. hostkey server cert sshkey.cert

Right now you have hostkey server before fibs

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-ssh.j2:57]: Screenshot 2024-03-01 at 1 11 23 PM

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-ssh.j2:60]: Mine is - 4.30.2F-33092737.4302F

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/management-ssh.j2:57]: Moved server_cert below client, as per the CLI

**Review by sugetha24 — APPROVED** [review]
LGTM
Tested:
Molecule test 
Documentation
Artifacts

**Review by JulioPDX — APPROVED** [review]
LGTM, thank you for the PR

**Pull request merged**